lib/SieveKeywordRegistry.php

```diff
@@ -73,15 +96,60 @@ public function __construct($extensions_enabled, $custom_extensions)
         }
     }
 
-    public function activate($extension)
+    /**
+     * Activates an extension.
+     *
+     * @param string $extension
+     */
+    public function activate(string $extension)
     {
         if (!isset($this->registry_[$extension]))
         {
             return;
         }
 
+        if (isset($this->loadedExtensions[$extension])) {
+            // already loaded
+            return;
+        } else {
+            $this->loadedExtensions[$extension] = true;
+        }
+        // we can safely unset the required extension
+        unset($this->requiredExtensions[$extension]);
+
+        if (isset($this->forbiddenExtensions[$extension])) {
```

Comment:
What do you think about doing this in `validateRequires`? Could centralize the validation a bit